### ISSUE TITLE
Question: Loss not decreasing

### ISSUE THREAD

**Issue by jonashaag** [Andong-Li-speech/RTNet#6]
Hi! I'd like to reproduce the very good results of your paper. When I run it on a 15,000 samples * 3s dataset, loss only decreases considerably at the very beginning (first few batches of epoch 1), and then stays the same. After 20 epochs of training, the results are not good. Any ideas what's wrong here? I have made no changes to the code except for what I submitted as PR, so using the default learning rate settings etc.

I'll upload a reconstruction sample after 20 epochs.

**Comment by jonashaag** [comment]
[out.wav.zip](https://github.com/Andong-Li-speech/RTNet/files/4592635/out.wav.zip)

**Comment by Andong-Li-speech** [comment]
> Hi! I'd like to reproduce the very good results of your paper. When I run it on a 15,000 samples * 3s dataset, loss only decreases considerably at the very beginning (first few batches of epoch 1), and then stays the same. After 20 epochs of training, the results are not good. Any ideas what's wrong here? I have made no changes to the code except for what I submitted as PR, so using the default learning rate settings etc.
> 
> I'll upload a reconstruction sample after 20 epochs.

Hi! First I want to appreciate your contributions as there is much room to improve the code quality. I think the problem is mainly due to self.Iter = 1. As we have to reuse the network multiple times.  self.Iter=1 means only one stage is used and no correlation can be established between adjacent stages. In addition, as we only give the supervision to the final stage, we should output the estimation result of the last stage during the decoding process.

**Comment by Andong-Li-speech** [comment]
> Hi! I'd like to reproduce the very good results of your paper. When I run it on a 15,000 samples * 3s dataset, loss only decreases considerably at the very beginning (first few batches of epoch 1), and then stays the same. After 20 epochs of training, the results are not good. Any ideas what's wrong here? I have made no changes to the code except for what I submitted as PR, so using the default learning rate settings etc.
> 
> I'll upload a reconstruction sample after 20 epochs.

Hi! I have uploaded the infos.zip, which contains the loss curves of different models and enhanced clips under -5dB SNR.  I hope it is helpful for you!
